src/Utilities/Compiler/Extensions/DiagnosticExtensions.cs

```diff
@@ -144,6 +154,29 @@ internal static class DiagnosticExtensions
                      messageArgs: args);
         }
 
+        public static Diagnostic CreateDiagnostic(
+            this SyntaxReference syntaxReference,
```

Comment:
I don't really like all these SyntaxReference extension methods, seem to be very specific to a set of internal analyzers in Roslyn.Diagnostic.Analyzers. Maybe it is better to move all of them to just that project inside a separate helper type?

Reply:
I am fine with this.